packages/gatsby/src/bootstrap/load-plugins/validate.ts

```diff
@@ -187,9 +188,9 @@ async function validatePluginsOptions(
   const newPlugins = await Promise.all(
     plugins.map(async plugin => {
       let gatsbyNode
-
       try {
-        gatsbyNode = require(`${plugin.resolve}/gatsby-node`)
+        const resolvedPlugin = resolvePlugin(plugin.resolve, rootDir)
```

Comment:
This breaks pnpm/yarn2 because it doesn't respect the location the plugin was sourced from